src/slurm_plugin/instance_manager.py

```diff
@@ -357,10 +251,14 @@ def _parse_nodes_resume_list(self, node_list: List[str]) -> defaultdict[str, def
         Sample NodeName: queue1-st-computeres1-2
         """
         nodes_to_launch = defaultdict(lambda: defaultdict(list))
+        logger.debug("Nodes already assigned to running instances: %s", self.nodes_assigned_to_instances)
         for node in node_list:
             try:
                 queue_name, node_type, compute_resource_name = parse_nodename(node)
-                nodes_to_launch[queue_name][compute_resource_name].append(node)
+                if node not in self.nodes_assigned_to_instances.get(queue_name, {}).get(compute_resource_name, []):
```

Comment:
NIT: This logic is because the job might have been submitted with non-exclusive node allocation? If so, maybe we can add a comment to mention this.

---

Optional:
- Can we perform this filtering in a different function in the node launching workflow? This function is meant to parse the resume list as it is.
- Alternatively we can rename this function to something like `get_nodes_to_launch_from_node_list`

Reply:
Thanks, I have added a comment for now.
There reason why I put the logic in the same method was in order to not redo the for loop once again. I will reconsider the implementation in next changes